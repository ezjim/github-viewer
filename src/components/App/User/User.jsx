import React from 'react';
import PropTypes from 'prop-types';

const User = ({ login, following, location, html_url, avatar_url }) => (
  <figure>
    <img src={avatar_url} alt={name} />
    <figcaption>
      <p>{login}</p>
      <p>{location}</p>
      <p>{following}</p>
      <p>{html_url}</p>
    </figcaption>
  </figure>
);
User.propTypes = {
  avatar_url: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  following: PropTypes.number.isRequired,
  location: PropTypes.string.isRequired,
  html_url: PropTypes.string.isRequired
};

export default User;
