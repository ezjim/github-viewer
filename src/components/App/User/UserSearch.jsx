import React from 'react';
import PropTypes from 'prop-types';

const UserSearch = ({ username, onUserChange, onUserSubmit }) => (
  <>
    <input type="text" name="username" value={username} onChange={onUserChange} />
    <input type="button" name="search" value="Search" onClick={onUserSubmit} />
  </>
);

UserSearch.propTypes = {
  username: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired,
  onUserSubmit: PropTypes.func.isRequired
};
export default UserSearch;
