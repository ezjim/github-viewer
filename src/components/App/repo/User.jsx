import React from 'react';
import PropTypes from 'prop-types';
const User = ({ name, followers, following, URL, avatar_url }) => (
  <figure>
    <img src={avatar_url} alt={name} />
    <figcaption>
      <p>{name}</p>
      <p>{followers}</p>
      <p>{following} </p>
      <p>{URL}</p>
    </figcaption>
  </figure>
);
User.propTypes = {
  avatar_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  URL: PropTypes.string.isRequired
};
export default User;
