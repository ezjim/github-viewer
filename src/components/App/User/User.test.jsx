import React from 'react';
import { shallow } from 'enzyme';
import User from './User.jsx';

describe('User component', () => {
  it('renders User', () => {
    const wrapper = shallow(<User
      avatar_url="https://avatars0.githubusercontent.com/u/56773448?v=4"
      login="ezjim"
      location="Portland"
      following={0}
      html_url="https://github.com/ezjim"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
