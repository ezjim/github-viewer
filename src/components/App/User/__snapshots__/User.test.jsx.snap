// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders User 1`] = `
<figure>
  <img
    alt=""
    src="https://avatars0.githubusercontent.com/u/56773448?v=4"
  />
  <figcaption>
    <p>
      ezjim
    </p>
    <p>
      Portland
    </p>
    <p>
      0
    </p>
    <p>
      https://github.com/ezjim
    </p>
  </figcaption>
</figure>
`;
